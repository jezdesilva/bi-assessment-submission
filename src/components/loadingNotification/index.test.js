import React from 'react';
import { shallow, mount } from 'enzyme';
import LoadingNotification from './index';

describe('LoadingNotification', () => {
  const loadingMessage = <div className="ui text loader">Loading...</div>;

  test('renders without crashing', () => {
    shallow(<LoadingNotification />);
  });

  test('should match the snapshot', () => {
    const wrapper = mount(<LoadingNotification show={true} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('when show is set to false, notification does not show', () => {
    const wrapper = mount(<LoadingNotification show={false} />);
    expect(wrapper.find('.content')).toHaveLength(0);
  });
  test('when show is set to true, notification does shows', () => {
    const wrapper = mount(<LoadingNotification show={true} />);
    expect(wrapper.find('.content')).toHaveLength(1);
  });
});

