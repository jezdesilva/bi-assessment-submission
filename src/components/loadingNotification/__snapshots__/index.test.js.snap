// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingNotification should match the snapshot 1`] = `"<div class=\\"ui active transition visible page dimmer\\" style=\\"display: flex;\\"><div class=\\"content\\"><div class=\\"ui text loader\\">Loading...</div></div></div>"`;
